modules/programs/fish.nix

```
@@ -12,6 +12,29 @@ let
     mapAttrsFlatten (k: v: "alias ${k} '${v}'") cfg.shellAliases
   );
 
+  envShellInit = pkgs.writeText "shellInit" cfge.shellInit;
+
+  envLoginShellInit = pkgs.writeText "loginShellInit" cfge.loginShellInit;
+
+  envInteractiveShellInit = pkgs.writeText "interactiveShellInit" cfge.interactiveShellInit;
+
+  fenv = pkgs.fish-foreign-env or pkgs.fishPlugins.foreign-env;
```

I ran into the same problem because of the redirect at `pkgs.fish-foreign-env`, looks like flipping around the packages would indeed fix it.

```suggestion
   fenv =  pkgs.fishPlugins.foreign-env or pkgs.fish-foreign-env;
```